ivy_tests/test_ivy/test_functional/test_core/test_linalg.py

```diff
@@ -1258,6 +1347,50 @@ def test_cholesky(
     )
 
 
+# cov
+@handle_test(
+    fn_tree="functional.ivy.cov",
+    dtype_x1_x2_cov=_get_dtype_value1_value2_cov(
+        available_dtypes=helpers.get_dtypes("numeric"),
+        min_num_dims=1,
+        max_num_dims=2,
+        min_dim_size=2,
+        max_dim_size=5,
+        min_value=1,
+        max_value=1e10,
+        abs_smallest_val=0.01,
+        large_abs_safety_factor=2,
+        safety_factor_scale="log",
+    ),
+)
+def test_cov(
+    *,
+    dtype_x1_x2_cov,
+    test_flags,
+    backend_fw,
+    fn_name,
+    on_device,
+    ground_truth_backend,
+):
+    dtype, x1, x2, rowVar, bias, ddof, fweights, aweights = dtype_x1_x2_cov
+    helpers.test_function(
+        ground_truth_backend=ground_truth_backend,
+        input_dtypes=dtype,
```

Comment:
please specify the dtype for all arrays according to sequence, eg:

in this test,
1st arg is x1
2nd arg is x2
3rd arg is fweights
4th arg is aweights
(scalar like int or bool are skipped)

so, the input_dtypes should be `[dtype[0], dtype[0], 'int64', 'float64']`

this will prevent the test function from assuming all array arguments follow the same dtype and cast them all to that dtype

this solution should work! I've tried it out myself on `numpy` backend and the test is passing!